concent_api/conductor/tasks.py

```diff
@@ -0,0 +1,76 @@
+
+from .models            import VerificationRequest
+from .models            import UploadReport
+from .models            import UploadRequest
+
+
+#@shared_task
+def verification_order_task(
+    subtask_id:             str,
+    src_code:               str,
+    extra_data:             str,
+    short_description:      str,
+    working_directory:      str,
+    performance:            str,
+    docker_images:          str,
+    source_file:            str,
+    result_file:            str,
+):
+    pass
+
+
+#@shared_task
+def verification_request_task(
+    compute_task_def:   message.ComputeTaskDef,
+    files:              list,
+):
+    assert isinstance(compute_task_def, message.ComputeTaskDef)
+
+    # The app creates a new instance of VerificationRequest.
+    verification_request = VerificationRequest(
+        subtask_id          = compute_task_def['subtask_id'],
+        src_code            = compute_task_def['src_code'],
+        extra_data          = compute_task_def['extra_data'],
+        short_description   = compute_task_def['short_description'],
+        working_directory   = compute_task_def['working_directory'],
+        performance         = compute_task_def['performance'],
+        docker_images       = compute_task_def['docker_images'],
+    )
+    verification_request.full_clean()
+    verification_request.save()
+
+    # The app creates ne UploadRequest for each file listed in the task.
+    for file in files:
+        upload_request = UploadRequest(
+            path                 = file,
+            verification_request = verification_request,
+        )
+        upload_request.full_clean()
+        upload_request.save()
+
+        # If there are already UploadReports corresponding to some files, the app links them together by setting the
+        # value of the foreign key in UploadRequest.
+        UploadReport.objects.filter(
+            path = file
+        ).update(
+            upload_request = upload_request
+        )
+
+    verification_request.refresh_from_db()
+
+    # If all expected files have been uploaded, the app sends verification_order task to the work queue.
+    if all([upload_request.upload_reports.exists() for upload_request in verification_request.upload_requests.all()]):
+        verification_order_task(  # TODO: Add .delay() when celery config is added
+            subtask_id          = verification_request.subtask_id,
+            src_code            = verification_request.src_code,
+            extra_data          = verification_request.extra_data,
+            short_description   = verification_request.short_description,
+            working_directory   = verification_request.working_directory,
+            performance         = verification_request.performance,
+            docker_images       = verification_request.docker_images,
+            source_file         = '??', # TODO: What should be passed here ?
+            result_file         = '??', # TODO: What should be passed here ?
```

Comment:
I'm going to update the issues today. Expecially two changes that will help you here:
- I'll specify more precisely when and how we go from a list of files to two specific source and result archive paths.
- Most of the fields here that come from `ComputeTaskDef` are going to be removed. On Tuesday I talked about it with Łukasz and Golem is going to change `TaskToCompute` a bit to make these fields more general and less specific to how Golem runs Blender as a desktop app. Until then we can igore them.

By the way, if you have such questions during implementation please always make a comment in the original issue to let me know. Maybe we should make it a rule. I won't always be able to find them in the code.

Reply:
:+1: Ok, looking forward to it, and I was going to ask those questions today, after I was sure I can't find answers in descriptions.

Reply:
Sorry about the delay. There's always something getting in the way I did not manage to get to this yet.

Reply:
I have updated the blueprint and all issues. You'll get these file names in `blender_verification_request`.